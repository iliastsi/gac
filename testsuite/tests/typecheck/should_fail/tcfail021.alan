ForwardDeclaration() : proc
  external(x : reference byte, y : int) : int;
  internal(y : int) : int;
  internal(y : int) : proc;
  foo() : proc;

  foo2() : proc
    external(x : reference byte, y : int) : proc;
    internal() : proc
    { }
    foo() : int;
  { foo(); external(42); }

  foo() : proc
  { }

  internal(x : byte) : proc
  { }
{
  foo();
}
