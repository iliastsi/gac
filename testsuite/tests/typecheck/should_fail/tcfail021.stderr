tcfail021.alan:4:3-10: Error: Conflicting definitions for `internal'
	Bound at: tcfail021.alan:3:3-10
	          tcfail021.alan:4:3-10

tcfail021.alan:8:5-12: Error: Conflicting definitions for `external'
	Bound at: tcfail021.alan:2:3-10
	          tcfail021.alan:8:5-12

tcfail021.alan:12:12-23: Error: Type mismatch at `external(42)'
	The function `external' is applied to 1 parameters but its type has 2

tcfail021.alan:17:3-10: Error: Declaration for `internal' doesn't match prototype
	Bound at: tcfail021.alan:4:3-10
	          tcfail021.alan:17:3-10

